config/confighttp/confighttp_test.go

```diff
@@ -61,3 +62,62 @@ func TestInvalidPemFile(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPServerSettingsError(t *testing.T) {
+	tests := []struct {
+		settings HTTPServerSettings
+		err      string
+	}{
+		{
+			err: "^failed to load TLS config: failed to load CA CertPool: failed to load CA /doesnt/exist:",
+			settings: HTTPServerSettings{
+				Endpoint: "",
+				TLSSetting: &configtls.TLSServerSetting{
+					TLSSetting: configtls.TLSSetting{
+						CAFile: "/doesnt/exist",
+					},
+				},
+			},
+		},
+		{
+			err: "^failed to load TLS config: for auth via TLS, either both certificate and key must be supplied, or neither",
+			settings: HTTPServerSettings{
+				Endpoint: "",
+				TLSSetting: &configtls.TLSServerSetting{
+					TLSSetting: configtls.TLSSetting{
+						CertFile: "/doesnt/exist",
+					},
+				},
+			},
+		},
+		{
+			err: "^failed to load TLS config: failed to load client CA CertPool: failed to load CA /doesnt/exist:",
+			settings: HTTPServerSettings{
+				Endpoint: "",
+				TLSSetting: &configtls.TLSServerSetting{
+					ClientCAFile: "/doesnt/exist",
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.err, func(t *testing.T) {
+			_, err := test.settings.ToListener()
+			assert.Regexp(t, test.err, err)
+		})
+	}
+}
+
+func ExampleHTTPServerSettings() {
```

Comment:
What's this function for? Doesn't seem to be used in any tests

Reply:
it is a testable example, will show in godoc https://blog.golang.org/examples